hooks/pre/30-upgrade.rb

```diff
@@ -60,15 +68,20 @@ def fail_and_exit(message)
   upgrade_step :stop_services
   upgrade_step :start_databases
 
-  if Kafo::Helpers.module_enabled?(@kafo, 'katello') || @kafo.param('capsule', 'pulp').value
+  if Kafo::Helpers.module_enabled?(@kafo, 'katello') || @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
     upgrade_step :migrate_pulp
     upgrade_step :start_httpd
   end
 
+  if @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
+    upgrade_step :remove_nodes_importers
```

Comment:
To make what's happening clearer, I might move this to the block above, remove the katello condition, and also add migrate_pulp, start_httpd in the katello if block below.

It's slightly less DRY but more readable. One can look and easily see "All these happen on the capsule", and "All those happen on the katello".